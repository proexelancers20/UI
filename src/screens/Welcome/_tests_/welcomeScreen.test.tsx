import React from 'react';
import WelcomeScreen from '..';
import { render } from '@testing-library/react-native';

jest.mock('react-native-size-matters',() => ({
    scale: jest.fn(),
    verticalScale: jest.fn()
}));

const component = (
  <WelcomeScreen
    navigation={{ navigate: jest.fn(), goBack: jest.fn(), canGoBack: jest.fn() } as any}
    route={undefined as any }
  />
);

describe('Welcome Screen', () => {
  it('render snap welcome screen', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot()
  })
});
