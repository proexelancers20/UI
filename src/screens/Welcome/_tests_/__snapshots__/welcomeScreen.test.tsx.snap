// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen render snap welcome screen 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "flex": 2,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/welcome.png",
        }
      }
      style={
        {
          "height": undefined,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginHorizontal": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FAA8D1",
            "borderRadius": undefined,
            "height": undefined,
            "justifyContent": "center",
          },
          {
            "marginBottom": undefined,
          },
        ]
      }
      testID="Welcome_sign_up_button"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "700",
            },
            undefined,
          ]
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FAA8D1",
            "borderRadius": undefined,
            "height": undefined,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#fff",
            "borderColor": "#000000",
            "borderWidth": 2,
          },
        ]
      }
      testID="Welcome_create_profile_button"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "700",
            },
            undefined,
          ]
        }
      >
        Create your profile
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column-reverse",
          "marginBottom": undefined,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontFamily": "NunitoSans_10pt-SemiBold",
            "fontSize": 16,
            "fontWeight": "500",
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      >
        Terms & Privacy Policy
      </Text>
    </View>
  </View>
</View>
`;
